Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -801,10 +809,37 @@ extension RestRequest: URLSessionDelegate {
             completionHandler(.performDefaultHandling, nil)
             return
         }
-
+        
         let warning = "Attempting to establish a secure connection; This is only supported by macOS 10.6 or higher. Resorting to default handling."
 
         switch (method, host) {
+        case (NSURLAuthenticationMethodClientCertificate, baseHost):
+            #if !os(Linux)
+            guard let certificatePath = self.certificatePath else {
```

Comment:
I think we should `guard let clientCertificateName = self.clientCertificateName` too, just in case someone passes us a path but no name?